### PR TITLE
fix: merge external default import

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
### Description
1. part of https://github.com/rolldown/rolldown/issues/3199
2. will try to merge two `external` default import if they are in same chunk

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
* **#3203** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/3203?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/3203?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: crates/rolldown/tests/esbuild/default/import_forms_with_minify_identifiers_and_no_bundle/artifacts.snap [hunk 1]
```diff
@@ -117,15 +117,15 @@ snapshot_kind: text
 ## entry.js
 
 ```js
-import * as ns2 from "foo";
 import * as ns from "foo";
-import def, { a, a2, b, default as def2, default as def3 } from "foo";
+import * as ns2 from "foo";
```
Comment (IWANABETHATGUY) [inline crates/rolldown/tests/esbuild/default/import_forms_with_minify_identifiers_and_no_bundle/artifacts.snap:121]: the order just follows the orignal span of the specifier, https://github.com/rolldown/rolldown/blob/8cf601e7c4ed0589c4ffaa63c8276afccd22a2ed/crates/rolldown/tests/esbuild/default/import_forms_with_minify_identifiers_and_no_bundle/entry.js?plain=1#L3-L6

File: crates/rolldown/src/stages/link_stage/bind_imports_and_exports.rs [hunk 2]
```diff
@@ -424,7 +443,7 @@ impl BindImportsAndExportsContext<'_> {
       let is_external = matches!(self.index_modules[rec.resolved_module], Module::External(_));
       if is_esm && is_external {
         if let Specifier::Literal(ref name) = named_import.imported {
-          if name.as_str() != "default" && is_validate_identifier_name(name) {
+          if is_validate_identifier_name(name) {
```
Comment (hyf0) [inline crates/rolldown/src/stages/link_stage/bind_imports_and_exports.rs:446]: Are we going to to support non valid identifier in the future?

File: crates/rolldown/tests/esbuild/default/import_forms_with_minify_identifiers_and_no_bundle/bypass.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # Reason
 1. Could be done in minifier
+2. Rolldown not try to merge external default import binding
```
Comment (hyf0) [inline crates/rolldown/tests/esbuild/default/import_forms_with_minify_identifiers_and_no_bundle/bypass.md]: Is this a typo? Seems this PR is gonna to support merging external default import binding.

**Review by hyf0 — APPROVED** [review]

File: crates/rolldown/tests/esbuild/default/import_forms_with_minify_identifiers_and_no_bundle/bypass.md (hunk 3 above)
Comment (IWANABETHATGUY) [reply crates/rolldown/tests/esbuild/default/import_forms_with_minify_identifiers_and_no_bundle/bypass.md]: Yeah, it is typo

File: crates/rolldown/src/stages/link_stage/bind_imports_and_exports.rs (hunk 2 above)
Comment (IWANABETHATGUY) [reply crates/rolldown/src/stages/link_stage/bind_imports_and_exports.rs:446]: in theory yes.

**Pull request merged**